src/main.ts

```diff
@@ -13,12 +13,17 @@ import { StatusRegistry } from './StatusRegistry';
 import { EditorSuggestor } from './Suggestor/EditorSuggestorPopup';
 import { StatusSettings } from './Config/StatusSettings';
 import type { Task } from './Task';
+import { TasksApi } from './TasksApi';
 
 export default class TasksPlugin extends Plugin {
     private cache: Cache | undefined;
     public inlineRenderer: InlineRenderer | undefined;
     public queryRenderer: QueryRenderer | undefined;
 
+    get api(): ReturnType<typeof TasksApi.GetApi> {
```

Comment:
What's the purpose of the `ReturnType` hoops please?

Reply:
Hi @claremacrae - nice to get feedback so fast.

> And some documentation would be greatly appreciated. With something this advanced, I suspect it may have a bit of an impact on user support load, and so I would be keen for the documentation to be thorough, taking people through using it - and recording any known limitations, for example.
> 
> I am sure you would, but it would really help for the docs to include a text version of anything the user would need to type in to a dialog, to help minimise support requests that turn out to be typos.

I will try to come up with some documentation. 
Where would be the best place this? In the [Advanced](https://obsidian-tasks-group.github.io/obsidian-tasks/advanced/) section or even as a new section?


> I am not sure that there is a way. If a test file could at least try and import the API, that would be start I suppose.

I will try to come up with some test that at least imports API.
 
> In the [Tasks docs](https://obsidian-tasks-group.github.io/obsidian-tasks/getting-started/create-or-edit-task/) it is referred to the as the ‘Create or edit Task’ Modal.
> 
> So perhaps createOrEditTask() - but then that begs the question: what happens if the user has a cursor already on a line of text? In Tasks, that line would be loaded in to the modal, for them to add data too.
> 
> Do you think people would expect that behaviour?

The current implementation only allows to retrieve the markdown string for a new task - editing an existing markdown string is not supported. Although this could be easily added.
`ShowUi` does not add a new task to any document, so I would not call the function like anything that could imply that (e.g. `createTask()`).

Some possible names that come to mind... `getTaskFromUI()` `openTaskUI()` `getTaskModal()` 

> What if the cursor was on a current task. Would they expect that task to be loaded?

The code is called from scripts. If the user wanted to edit the task the cursor is on, he/she would be responsible to retrieve the text of the current line and give the string as a parameter to the function (if we allow editing).

Reply:
Just being explicit. Will remove the redundant `ReturnType`.

Reply:
> Just being explicit. Will remove the redundant `ReturnType`.

Oh OK. I had assumed it was useful...